File: js/020_ValidParentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    var cs = [];
    for(var i in s){
        switch( s[i] ){
            case '(':
            case '[':
            case '{':
                cs.push(s[i]);
                break;
            case ')':
                if(cs.length < 1 || cs.pop() !== '(') return false;
                break;
            case ']':
                if(cs.length < 1 || cs.pop() !== '[') return false;
                break;
            case '}':
                if(cs.length < 1 || cs.pop() !== '{') return false;
                break;
        }
    }
    return cs.length === 0;
};

File: cpp/020_ValidParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> cStack;
        for(auto i : s){
            switch(i){
                case '(':
                case '[':
                case '{':
                    cStack.push(i);
                    break;
                case ')':
                    if(cStack.empty() || cStack.top() != '(') return false;
                    cStack.pop();
                    break;
                case ']':
                    if(cStack.empty() || cStack.top() != '[') return false;
                    cStack.pop();
                    break;
                case '}':
                    if(cStack.empty() || cStack.top() != '{') return false;
                    cStack.pop();
                    break;
                default:
                    return false;
            }
        }
        return cStack.empty();
    }
};